goalgorilla/open_social
# Issue #3385634 by nkoporec: Inline images are not displaying in social footer block

## Problem
When inserting an inline image in social footer block via the admin/config/opensocial/footer-block settings page, the image is not rendered after hitting the save button. The DOM element is there but the URL to the image is broken for other users (it works for the one that saved the form and uploaded the image).

The reason for this is that by default CKEditor will save inline images in the private directory, this ties the image to be only viewed/accessed by users who created this private image or have access to the entity that this image is attached to, since this form is not tied to any entity but rather its a custom implementation it means that only user who uploaded the inline image will have access to it.

## Solution
Implement a hook_file_download() in social_footer module that allows access to private images used within the text field of the social footer block settings.

## Issue tracker
https://www.drupal.org/project/social/issues/3385634

## Theme issue tracker
*[Required if applicable] Paste a link to the drupal.org theme issue queue item, either from [socialbase](https://www.drupal.org/project/socialbase) or [socialblue](https://www.drupal.org/project/socialblue). If any other issue trackers were used, include links to those too.*

## How to test
1. Log in as user 1.
2. Go to admin/config/opensocial/footer-block and upload an image via the CKEditor.
3. Save the form
4. Log in as different user and check the footer block if the image is displaying correctly.

## Definition of done
### Before merge

### After merge

## Screenshots
Before:
![image](https://github.com/goalgorilla/open_social/assets/35064680/224381b2-be3e-4160-a800-9fe99aa166bc)


After:
![image](https://github.com/goalgorilla/open_social/assets/35064680/97f44d1c-9595-472f-8b74-96d515a35e17)


## Release notes
Fixed an issue where inline images would not render correctly in social footer block.

## Change Record


## Translations
